### PR TITLE
Add provider configuration documentation

### PR DESCRIPTION
**Pull request by danielrbradley**

### CODE REVIEW

**Review by squaremo — APPROVED** [review]
Useful additional info to have in there :+1:

File: themes/default/content/docs/intro/concepts/config.md [hunk 1]
```diff
@@ -284,7 +284,20 @@ public static void stack(Context ctx) {
 
 {{< /chooser >}}
 
-## Special Configuration Options
+## Provider Configuration Options
+
+There are three ways to configure providers:
+
+1. Set configuration keys in the stack configuration file: `pulumi config set [PROVIDER]:[KEY] [VALUE]`
```
Comment (lukehoban) [inline themes/default/content/docs/intro/concepts/config.md:291]: Throughout the rest of this document - this is referred to as "Pulumi configuration" but here we use the term "stack configuration file". Suggest we continue to use "Pulumi configuration" in this section for consistency.

File: themes/default/content/docs/intro/concepts/config.md [hunk 2]
```diff
@@ -284,7 +284,20 @@ public static void stack(Context ctx) {
...
+2. Set a provider-specific environment variable
+3. Pass arguments to the provider's SDK constructor, in your program
+
+Please note:
+
+* Configuration file settings are only used by the default provider. If you instantiate a provider object, it will not read values from the stack configuration.
```
Comment (lukehoban) [inline themes/default/content/docs/intro/concepts/config.md:297]: Cross link "provider" to the documentation on Provider's (and possibly capitalize and format as code `Provider`?)

Cross link default provider to documentation on default providers?

File: themes/default/content/docs/intro/concepts/config.md [hunk 3]
```diff
@@ -284,7 +284,20 @@ public static void stack(Context ctx) {
...
+* The precedence of configuration sources (configuration file, environment and args) can vary between providers. Please refer to the provider's documentation for specific configuration instructions.
```
Comment (lukehoban) [inline themes/default/content/docs/intro/concepts/config.md:298]: Perhaps cross link to the registry?

File: themes/default/content/docs/intro/concepts/config.md [hunk 4]
```diff
@@ -284,7 +284,20 @@ public static void stack(Context ctx) {
...
+
```
Comment (lukehoban) [inline themes/default/content/docs/intro/concepts/config.md:299]: Is there an example that would be useful here?  Possibly not - especially if cross linking works to send folks to content that does have examples.

**Pull request merged**